internal/providers/terraform/aws/instance.go

```diff
@@ -31,16 +35,40 @@ func NewInstance(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
 		purchaseOption = "spot"
 	}
 
+	var instanceType, ami, cpuCredits string
+	var ebsOptimized, monitoring bool
+
+	launchTemplateRefs := d.References("launch_template.#.id")
+	if len(launchTemplateRefs) == 0 {
+		launchTemplateRefs = d.References("launch_template.#.name")
+	}
+
+	if len(launchTemplateRefs) > 0 {
+		ref := launchTemplateRefs[0]
+
+		instanceType = ref.Get("instance_type").String()
+		ami = ref.Get("image_id").String()
+		ebsOptimized = ref.Get("ebs_optimized").Bool()
+		monitoring = ref.Get("monitoring.0.enabled").Bool()
+		cpuCredits = ref.Get("credit_specification.0.cpu_credits").String()
+	}
+
+	instanceType = d.GetStringOrDefault("instance_type", instanceType)
+	ami = d.GetStringOrDefault("ami", ami)
+	ebsOptimized = d.GetBoolOrDefault("ebs_optimized", ebsOptimized)
+	monitoring = d.GetBoolOrDefault("monitoring", monitoring)
+	cpuCredits = d.GetStringOrDefault("credit_specification.0.cpu_credits", cpuCredits)
```

Comment:
If these attributes are defined on the instance resource they override the launch template's values.